src/main/java/scala_maven/ScalaCompilerSupport.java

```diff
@@ -97,10 +97,14 @@ protected int compile(List<File> sourceRootDirs, File outputDir, File analysisCa
         List<String> classpathElements, boolean compileInLoop) throws Exception {
         if (!compileInLoop && recompileMode == RecompileMode.incremental) {
             // if not compileInLoop, invoke incrementalCompile immediately
-            return incrementalCompile(classpathElements, sourceRootDirs, outputDir, analysisCacheFile, false);
+            long n0 = System.nanoTime();
+            int res = incrementalCompile(classpathElements, sourceRootDirs, outputDir, analysisCacheFile, false);
+            getLog().info(String.format("compile in %.1f s", (System.nanoTime() - n0) / 1_000_000_000.0));
+            return res;
```

Comment:
instead of return,  I prefer to have else and to wrap the if & else block like 
```
long n0=...
if ....
else ...
long durationMs = (System.nanoTime() - n0) / 1_000_000.0;
getLog().info(String.format("compile in %.1f s", durationMs / 1_000.0));
return files.size();
```

But it should require to rewrite / to refactor the other return :-(
Maybe remove the log / split about "prepare-compile" (TBH I don't remember why we split the duration and made the computation of lastCompileAt in the middle).

Reply:
I didn't want to refactor the whole thing. At first glance, it looked pretty complicated to refactor because of the compilation loop feature (that I've never used).